folder-structure-and-file-naming.md

```diff
@@ -0,0 +1,123 @@
+│   └── settings.json           // (SHARED) VSCode settings, that we share across team members (Spell Checker, formatter, etc.).
+│
+├── android                     // Contains Android-specific files.
+├── ios                         // Contains iOS-specific files.
+├── node_modules                // Contains Node Modules
+├── src                         // Contains source files & folders
+│   ├── assets                  // Contains assets files. May be different for each project.
+│   │   ├── fonts
+│   │   ├── icons
+│   │   └── images
+│   │
+│   ├── components              // Contains React Native components, that we use for creating screens
+│   ├── constants               // Contains files, from which we can import constants across the app
+│   ├── navigation              // Contains Navigation-specific files.
+│   ├── screens                 // Contains screens, that have some logic and builds from components
+│   ├── models                  // Contains folders & files that describe TS interfaces
+│   ├── store                   // Contains Redux-specific files.
+│   │   ├── actions             // Redux Actions
+│   │   ├── reducers            // Redux Reducers
+│   │   ├── middlewares         // Redux Middlewares
+│   │   ├── rootReducer.js      // Combines all reducers
+│   │   └── store.js            // Entry-point of the Redux Store
+│   │
+│   ├── theme                   // Contains theme files for styling the App
+│   ├── utils                   // Contains helper function
+│   └── App.js                  // Entry-point of the App in src folder
+│ 
+├── .buckconfig
+├── .eslintrc.js
+├── .prettierrc.js
+├── .gitattributes
+├── .gitignore
+├── .watchmanconfig
+├── app.json
+├── babel.config.js
+├── tsconfig.json               // or jsconfig.json
+├── index.js                    // Entry-point of the App
+├── package.json                // Metadata relevant to the project
+└── yarn.lock
+```
+
+<br/>
+<br/>
+
+# File Naming
+
+We are using `camelCase` for common folder names ('screens', 'components', 'assets', etc.)
+
+For React files `PascalCase`.
+
+> Note: By default, Mac OS is case-insensitive but case-preserving. This means that as far as macOS is concerned Test.js and test.js are the same file. So be attentive to the case when creating folders/files so that they don't run into that problem.
+
+
+Example:
+
+```
+├── components
+│   ├── backButton
+│   │   ├── index.js
+│   │   ├── BackButton.js
+│   │   └── BackButton-styles.js
+```
+
+Where index.js will be:
+
+```js
+import BackButton from './BackButton';
+
+export default BackButton;
+```
+
+And BackButton.js will be:
+
+```js
+// Libs
+import React from 'react';
+
+// Globals
+import BackIcon from 'assets/icons/back-icon.svg';
+
+// Locals
+import * as Styles from './BackButton-styles';
+
+const BackButton = ({ onPress }) => (
+  <Styles.BackButton onPress={onPress}>
+    <BackIcon />
+  </Styles.BackButton>
+);
```

Comment:
This assumes we will use styled-components, but we did not decide on that yet. It would be good to add a note about this. Something like: "This guide assumes styled-components library is used for styling, but that is not a requirement and should be changed in future once we make a decision on the way we do styling."